prereise/gather/winddata/hrrr/helpers.py

```diff
@@ -0,0 +1,56 @@
+from datetime import timedelta
+
+
+def dt_iter(start_dt, end_dt, delta=timedelta(days=1)):
+    """Iterates from a start datetime (inclusive) to a end datetime (exclusive)
+    with step size of delta
+
+    :param datetime.datetime start_dt: datetime to start at
+    :param datetime.datetime end_dt: datetime to end at
+    :param datetime.timedelta delta: delta to step increase with
+
+    :return: (*Generator[datetime.datetime]*)
+    """
+    while start_dt < end_dt:
+        yield start_dt
+        start_dt += delta
+
+
+def formatted_filename(dt, product, hours_forecasted):
+    """Deterministically returns a filename
+
+    :param datetime.datetime dt:
+    :param string product:
+    :param string hours_forecasted:
+
+    :return: (*str*)
+    """
+    return f"{dt.strftime('%Y_%m_%d_%Hhr')}_{product}_{hours_forecasted}.grib2"
+
+
+def generate_index_url(url):
+    """Adds .idx to end of a url
+
+    :param string url:
+
+    :return: (*str*)
+    """
+    return f"{url}.idx"
+
+
+def get_indices_that_contain_selector(input_list, selectors):
+    """Generates list of indices of strings in input_list that
+    contain a string inside of selectors
+
+    :param list input_list: list of strings
+    :param list selectors: list of strings
+
+    :return: (*list*)
+    """
+
+    return_list = []
+    for i, item in enumerate(input_list):
+        for selector in selectors:
+            if selector in item:
+                return_list.append(i)
```

Comment:
If multiple selectors are present in the same string in `input_list`, do we want that entry added multiple times, or do we want a `break` inside this innermost block?

Reply:
nice, went with @ahurlils suggestion